Tests.WebView/CommonTests.cs

```diff
@@ -0,0 +1,194 @@
+        public async Task UnhandledExceptionEventIsCalledOnListenerError() {
+            await Run(() => {
+                var taskCompletionSource = new TaskCompletionSource<bool>();
+                const string ExceptionMessage = "hey";
+                Exception exception = null;
+
+                WithUnhandledExceptionHandling(async () => {
+                    var listener = TargetView.AttachListener("event_name");
+                    listener.Handler += delegate {
+                        taskCompletionSource.SetResult(true);
+                        throw new Exception(ExceptionMessage);
+                    };
+
+                    await Load($"<html><script>{listener}</script><body></body></html>");
+                    await taskCompletionSource.Task;
+                    Assert.IsTrue(exception.Message.Contains(ExceptionMessage));
+                },
+                e => {
+                    exception = e;
+                    return true;
+                });
+            });
+        }
+
+        [Test(Description = "Before navigate hook is called")]
+        public async Task BeforeNavigateHookCalled() {
+            await Run(async () => {
+                var taskCompletionSource = new TaskCompletionSource<bool>();
+                var beforeNavigatedCalled = false;
+                TargetView.BeforeNavigate += (request) => {
+                    beforeNavigatedCalled = true;
+                    taskCompletionSource.SetResult(true);
+                    request.Cancel();
+                };
+                TargetView.Address = "https://www.google.com";
+                await taskCompletionSource.Task;
+                Assert.IsTrue(beforeNavigatedCalled);
+            });
+        }
+
+        [Test(Description = "Javascript evaluation on navigated event does not block")]
+        public async Task JavascriptEvaluationOnNavigatedDoesNotBlock() {
+            await Run(async () => {
+                var taskCompletionSource = new TaskCompletionSource<bool>();
+                var navigated = false;
+                TargetView.Navigated += delegate {
+                    TargetView.EvaluateScript<int>("1+1");
+                    navigated = true;
+                    taskCompletionSource.SetResult(true);
+                };
+                await Load("<html><body></body></html>");
+                await taskCompletionSource.Task;
+                Assert.IsTrue(navigated);
+            });
+        }
+
+        [Test(Description = "Setting zoom works as expected")]
+        [Ignore("Zoom not working in CefGlue")]
+        public async Task ZoomWorksAsExpected() {
+            await Run(async () => {
+                await Load("<html><body>Zoom text</body></html>");
+
+                const double Zoom = 1.5;
+                var zoomTask = Dispatcher.UIThread.InvokeAsync(() => TargetView.ZoomPercentage = Zoom);
```

Comment:
probably the zoom itself takes time to take effect...

Reply:
So, we should wait for the zoom property to be set instead?

Reply:
Delete this test ... it should be done at cefglue level... but does not have much value